CMakeLists.txt

```diff
@@ -1,73 +1,71 @@
-cmake_minimum_required(VERSION 3.16 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.16 FATAL_ERROR) # Policies <= CMP0097 default to NEW
 
-message(AUTHOR_WARNING "If the build fails, please try the autotools/qmake method.")
-
-if (POLICY CMP0074)
-    cmake_policy(SET CMP0074 NEW)
-endif()
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
 
 # TODO: fix the macOS bundle stuff
-if(POLICY CMP0080)
-    cmake_policy(SET CMP0080 OLD)
+cmake_policy(SET CMP0080 OLD)
+
+# this must be defined before project() or enable_language() to take effect
+if(WIN32)
+    if(GENERATOR_IS_MULTI_CONFIG)
+        set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<BOOL:${QBT_MSVC_RUNTIME_DYNAMIC}>:DLL>$<$<CONFIG:Debug>:Debug>")
```

Comment:
What does this do?
and if I am right in thinking what this do then you don't need to do it, cmake handles this itself.

Reply:
@jagannatharjun 
The thing is, if you use a multi-config generator, `CMAKE_BUILD_TYPE` is ignored. Conversely, if you use a single-config generator, `CONFIG` is not defined. (https://cmake.org/cmake/help/v3.16/variable/CMAKE_BUILD_TYPE.html)